### PR TITLE
sof_perf_analyzer: add skip-to-first-trace option

### PR DESCRIPTION
**Pull request by aiChaoSONG**
In CI test, some traces from previous test case will apprear in the mtrace of current test case, this is a know issue.

Previously, we always skip to the first valid trace of this test case, this could raise trouble for developer use of this script.

This patch adds skip-to-first-trace option, for CI test, this option should be set (True), and for developer use, this option should be left as it is (False).

### CODE REVIEW

**Review by singalsu — APPROVED** [review]
This patch fixes the problem that I had. Thank you very much!

File: tools/sof_perf_analyzer.py [hunk 1]
```diff
@@ -295,6 +296,10 @@ def parse_args():
                         help='Kernel message file captured with journalctl or other log utility')
     parser.add_argument('--out2csv', type=pathlib.Path, required=False,
                     help='Output SOF performance statistics to csv file')
+    parser.add_argument('-s', '--skip-to-first-trace', action="store_true",  default=False,
+                        help='''In CI test, some traces from previous test case will apprear in
```
Comment (marc-hb) [inline tools/sof_perf_analyzer.py:300]:
```suggestion
                        help='''In CI test, some traces from previous test case will appear in
```

File: tools/sof_perf_analyzer.py [hunk 2]
```diff
@@ -295,6 +296,10 @@ def parse_args():
...
+the mtrace of current test case, this flag is used to denote if we
+want to skip those traces, for CI test, set the flag to True''')
```
Comment (marc-hb) [inline tools/sof_perf_analyzer.py:302]: Please don't "hide" the actual logic behind abstract names. This is test code, we need as few abstractions as possible.
```suggestion
want to skip until the first line with a timestamp between 0 and 1s. For CI test, set the flag to True''')
```

File: tools/sof_perf_analyzer.py [hunk 3]
```diff
@@ -169,8 +165,13 @@ def process_trace_file():
...
         trace_item_gen = make_trace_item(file)
-        trace_prev = skip_to_first_trace(trace_item_gen)
-        if trace_prev is None:
+        trace_prev = None
+        try:
+            if args.skip_to_first_trace:
+                trace_prev = skip_to_first_trace(trace_item_gen)
+            else:
+                trace_prev = next(trace_item_gen)
+        except:
```
Comment (marc-hb) [inline tools/sof_perf_analyzer.py:174]: https://pylint.pycqa.org/en/latest/user_guide/messages/warning/broad-exception-caught.html

Why do you need to catch anything here? Just let the original exception propagate to the top. This is test code, we want as few abstractions and layers as possible when things go wrong.

**Review by marc-hb — CHANGES_REQUESTED** [review]
The pylint warning may go away if you drop the too broad except?

File: tools/sof_perf_analyzer.py (hunk 3 above)
Comment (aiChaoSONG) [reply tools/sof_perf_analyzer.py:174]: The raised exception has no useful information, I want to add some useful information here, I will use below code, which is better, add message to the exception, and raise it again.
```
        try:
            if args.skip_to_first_trace:
                trace_prev = skip_to_first_trace(trace_item_gen)
            else:
                trace_prev = next(trace_item_gen)
        except StopIteration as si:
            si.args = 'No valid trace in provided file',
            raise 
```

**Review by marc-hb — DISMISSED** [review]
Nit: could you add a blank line before 190?

Note: still a Github Draft

File: case-lib/lib.sh [hunk 4]
```diff
@@ -901,9 +901,9 @@ perf_analyze()
     dlogi "Checking SOF component performance"
     if [ -e "$LOG_ROOT/mtrace.txt" ]; then
         if [ -e "$LOG_ROOT/dmesg.txt" ]; then
-            perf_cmd="sof_perf_analyzer.py --kmsg=$LOG_ROOT/dmesg.txt --out2csv $LOG_ROOT/sof_perf.txt $LOG_ROOT/mtrace.txt "
+            perf_cmd="sof_perf_analyzer.py -s --kmsg=$LOG_ROOT/dmesg.txt --out2csv $LOG_ROOT/sof_perf.txt $LOG_ROOT/mtrace.txt "
         else
-            perf_cmd="sof_perf_analyzer.py --out2csv $LOG_ROOT/sof_perf.txt $LOG_ROOT/mtrace.txt"
+            perf_cmd="sof_perf_analyzer.py -s --out2csv $LOG_ROOT/sof_perf.txt $LOG_ROOT/mtrace.txt"
```
Comment (marc-hb) [inline case-lib/lib.sh:906]: Please use the long option so people don't have to look it up. Short options save repetitive typing but it's not a problem here.

**Review by marc-hb — APPROVED** [review]

**Comment by aiChaoSONG** [comment]
Thanks Marc, the test is good, see internal result 31419, merge

**Pull request merged**